server/service/src/sync/test/integration/errors.rs

```diff
@@ -77,8 +77,10 @@ mod tests {
             .update_sync_settings(&service_context, &sync_settings)
             .unwrap();
 
+        let hardware_id = service_provider.app_data_service.get_hardware_id().unwrap();
+
         let synchroniser =
-            get_synchroniser_with_hardware_id(&connection_manager, &sync_settings, "id1");
```

Comment:
This looks like a change in functionality - the test was failing on the post-initialisation sync with the "id1" hardware ID, because it already had one set (i.e. the error that is supposed to happen below) 

I'm not super familiar with all of it, but this test reads to me like we were expecting there to no hardware ID set at this step, so we'd set it to id1, then change to id2 and see the error. And now it's failing, because there is already a hardware ID by the time we get to this step. 

I've fixed it by querying for the hardware ID instead, but just wanting to double check this is expected/we just forgot to update the test, or if there are any other implications?

Reply:
Yep, we had issues with hardware_id when doing omSupply central integration test (since it was possible for omSupply central to be using the same site as used for configuring test data on mSupply), so we've changed to allow for this (https://github.com/msupply-foundation/open-msupply/pull/2819/files#diff-8e0beeb411e6b526997da24481dc9d6d0979116b644fa22ca67a4c2d59055ae1, under central_server_configurations.rs, and test_helplers).

Forgot to update some tests, not sure how they were passing.

I actually was suppose to fix them in a previous PR i made, Clemens found this error too, but i think i forgot to commit. Anyways, when i fixed, i just removed the first sync in that test (since the correct hardware_id would already be set during configuration of test site and test data)